AK/Concepts.h

```diff
@@ -6,11 +6,28 @@
 
 #pragma once
 
+#include <AK/Forward.h>
 #include <AK/IterationDecision.h>
 #include <AK/StdLibExtras.h>
 
 namespace AK::Concepts {
 
+template<typename T>
+struct __decay {
+    typedef Detail::RemoveCV<T> type;
+};
+template<typename T>
+struct __decay<T[]> {
+    typedef T* type;
+};
+template<typename T, size_t N>
+struct __decay<T[N]> {
+    typedef T* type;
+};
+// FIXME: Function decay
+template<typename T>
+using Decay = typename __decay<T>::type;
```

Comment:
I would expect this to be in StdLibExtraDetails.h, it's not really a concept.

Reply:
this requires size_t which is not available in StdLibExtraDetails

Reply:
Ah, unfortunate - it's a bit ugly but it looks like other traits either use `decltype(sizeof(T))` in place of `size_t`, or just `unsigned`.

Reply:
You could just use `auto N` instead of `size_t N`